### PR TITLE
Update Lookup Vulnerable Hosts from Slack to use Parameters

### PR DESCRIPTION
**Pull request by cmchugh-r7**
## Requirements

This updates the workflow to use parameters for configuring slack channel name, max hosts, and max vulnerabilities.

### Checklist
- [x] Workflow verified as working in a test
- [x] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [x] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Workflow includes a detailed Description
- [x] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Triggers, Decisions, and other workflow steps include descriptions where needed
- [x] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [x] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [x] help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [x] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

Screen Shot 2021-02-12 at 9 21 46 AM

Screen Shot 2021-02-16 at 3 15 53 PM

#### Workflow Builder

workflow

#### Job

Screen Shot 2021-02-12 at 9 22 55 AM

#### Artifact

Screen Shot 2021-02-12 at 9 23 26 AM

### CODE REVIEW

**Comment by cmchugh-r7** [comment]
Waiting on finalized language for parameter setup instructions, then will update help+description. Cannot be merged til param release.

File: workflows/Lookup_Vulnerable_Hosts_From_Slack/Lookup_Vulnerable_Hosts_From_Slack.icon [hunk 1]
```diff
@@ -3950,16 +3860,37 @@
...
+        "parameters": {
+          "definitionSchema": {
+            "type": "object",
+            "required": [],
+            "properties": {
+              "max hosts": {
+                "type": "integer",
+                "description": ""
+              },
+              "channel name": {
```
Comment (malolans-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Slack/Lookup_Vulnerable_Hosts_From_Slack.icon:3873]: Should we have descriptions for the parameters?

Comment (cmchugh-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Slack/Lookup_Vulnerable_Hosts_From_Slack.icon]: added!

File: workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md [hunk 2]
```diff
@@ -67,6 +72,7 @@ _There is no troubleshooting information at this time_
 
 # Version History
 
+* 2.0.0 - Use Workflow Parameters to configure Slack channel name, max vulnerabilities, and max hosts values (and remove the Settings step); Update CSV to version 1.1.6; Update Type Converter to 1.7.0
```
Comment (jschipp-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md:75]: Style is `|` in the Changelog 

```suggestion
* 2.0.0 - Use Workflow Parameters to configure Slack channel name, max vulnerabilities, and max hosts values (and remove the Settings step) | Update CSV to version 1.1.6 | Update Type Converter to 1.7.0
```

Comment (cmchugh-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md]: ahh thanks Jon

**Comment by cmchugh-r7** [comment]
Validator is failing due to diff between description and spec, however as per [this thread](https://rapid7.slack.com/archives/C6GJ6F420/p1613506809051900) this check is getting removed and we should keep the "short" description in the yaml

File: workflows/Lookup_Vulnerable_Hosts_From_Slack/Lookup_Vulnerable_Hosts_From_Slack.icon [hunk 3]
```diff
@@ -3950,16 +3860,37 @@
...
-        "humanCostDisplayUnit": ""
+        "humanCostDisplayUnit": "",
+        "parameters": {
+          "definitionSchema": {
+            "type": "object",
+            "required": [],
+            "properties": {
+              "Max Hosts": {
+                "type": "integer",
+                "description": "The maximum number of hosts listed per vulnerability (recommended default is 10)"
```
Comment (jnikitin-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Slack/Lookup_Vulnerable_Hosts_From_Slack.icon:3871]: I was maybe thinking of adding "affected hosts", but idk if the change is really needed.

File: workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md [hunk 4]
```diff
@@ -24,9 +24,14 @@ Sample Slack command:
...
 
-Once the workflow has been imported, **the first Slack step will need the channel name updated to suit your Slack environment!** Edit the input with the preset text of `change_me` in first Slack step in the workflow. There is also an optional Settings step which can be edited to change the maximum number of vulnerabilities and maximum hosts listed per vulnerability which are returned. The defaults are 50 vulnerabilities and 10 assets per vulnerability.
+This workflow leverages InsightConnect's Parameters feature. This feature allows variables used multiple times throughout a workflow to be entered once and then referenced throughout the workflow.
 
-After configuring the Slack and Setting steps, activate the workflow in order to trigger it.
+There are three parameters you will need to configure in order to complete setup of your workflow:
+* `Slack Channel`:  the Slack channel name in your environment where the workflow should be triggered and respond
```
Comment (jnikitin-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md:30]: two spaces here

File: workflows/Lookup_Vulnerable_Hosts_From_Slack/Lookup_Vulnerable_Hosts_From_Slack.icon (hunk 3 above)
Comment (cmchugh-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Slack/Lookup_Vulnerable_Hosts_From_Slack.icon:3871]: ah thats a good idea, i like that, i'll update

File: workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md [hunk 5]
```diff
@@ -24,9 +24,14 @@ Sample Slack command:
...
 
-Once the workflow has been imported, **the first Slack step will need the channel name updated to suit your Slack environment!** Edit the input with the preset text of `change_me` in first Slack step in the workflow. There is also an optional Settings step which can be edited to change the maximum number of vulnerabilities and maximum hosts listed per vulnerability which are returned. The defaults are 50 vulnerabilities and 10 assets per vulnerability.
+This workflow leverages InsightConnect's Parameters feature. This feature allows variables used multiple times throughout a workflow to be entered once and then referenced throughout the workflow.
 
-After configuring the Slack and Setting steps, activate the workflow in order to trigger it.
+There are three parameters you will need to configure in order to complete setup of your workflow:
+* `Slack Channel`: the Slack channel name in your environment where the workflow should be triggered and respond
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md:30]:
```suggestion
* `Slack Channel`: The Slack channel name in your environment where the workflow should be triggered and respond
```

File: workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md [hunk 6]
```diff
@@ -24,9 +24,14 @@ Sample Slack command:
...
+* `Max Hosts`: the maximum number of hosts listed per vulnerability (recommended default is 10)
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md:31]:
```suggestion
* `Max Hosts`: The maximum number of hosts listed per vulnerability (recommended default is 10)
```

File: workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md [hunk 7]
```diff
@@ -24,9 +24,14 @@ Sample Slack command:
...
+* `Max Vulnerabilities`: the maximum number of returned vulnerabilities (recommended default is 50)
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md:32]:
```suggestion
* `Max Vulnerabilities`: The maximum number of returned vulnerabilities (recommended default is 50)
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

File: workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md (hunk 7 above)
Comment (cmchugh-r7) [reply workflows/Lookup_Vulnerable_Hosts_From_Slack/help.md]: all updated!

**Review by jschipp-r7 — DISMISSED** [review]

**Review by smorrison-r7 — CHANGES_REQUESTED** [review]
Aligning Version Formatting

File: workflows/Lookup_Vulnerable_Hosts_From_Slack/workflow.spec.yaml [hunk 8]
```diff
@@ -3,13 +3,13 @@ products: ["insightconnect", "insightvm"]
 name: Lookup_Vulnerable_Hosts_From_Slack
 title: Lookup Vulnerable Hosts from Slack
 description: Lookup vulnerability details, including a summary of the vulnerability, the affected software package, the remediation or mitigation step, and a list of all affected hosts in the user's environment.
-version: 1.1.0
+version: 2.0.0
```
Comment (mrinehart-r7) [inline workflows/Lookup_Vulnerable_Hosts_From_Slack/workflow.spec.yaml:6]:
```suggestion
version: 1.2.0
```
Mismatch between changelog and this version

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — DISMISSED** [review]

**Review by smorrison-r7 — APPROVED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Review by malolans-r7 — APPROVED** [review]

**Pull request merged**